std/file.d

```diff
@@ -3902,6 +3903,15 @@ string tempDir() @trusted
             DWORD len = GetTempPathW(buf.length, buf.ptr);
             if (len) cache = toUTF8(buf[0 .. len]);
         }
+        else version(Android)
+        {
+            // Don't check for a global temporary directory as
+            // Android doesn't have one.  Instead, use the local
+            // directory or set cache here to your app's local
+            // directory.
+
+            cache = getcwd();
```

Comment:
Would you prefer a `goto` to the last line? ;) The whole point of the cache is that this is only set once, so I doubt you could optimize it further. :)